src/components/all-on-one-form-sub-element-button/all-on-one-form-sub-element-button.ts

```diff
@@ -11,37 +11,27 @@ export class AllOnOneFormSubElementButtonComponent {
   @Input() sectionsToShow: string;
   serious: boolean = false;
   dangerous: boolean = false;
+  faultCount: number = 0;
 
   constructor(private faultStore: FaultStoreProvider) {}
 
   ngOnChanges() {
-    const controlKey = 'manoeuver' + this.sectionsToShow + 'Controlv1';
-    const obsKey = 'manoeuver' + this.sectionsToShow + 'Observationv1';
-
     this.faultStore.currentfaults$.subscribe((data) => {
-      if (data[obsKey]) {
-        if (data[obsKey].serious) {
-          this.serious = !!data[obsKey].serious;
-        }
-        if (data[obsKey].dangerous) {
-          this.dangerous = !!data[obsKey].dangerous;
-        }
-      }
-
-      if (data[controlKey]) {
-        if (data[controlKey].serious) {
-          this.serious = !!data[controlKey].serious;
-        }
-
-        if (data[controlKey].dangerous) {
-          this.dangerous = !!data[controlKey].dangerous;
+      let serious = false;
+      let dangerous = false;
+      let sumCount = 0;
+
+      Object.keys(data).map((section) => {
+        if (section.includes(this.sectionsToShow) && this.sectionsToShow !== '') {
+          sumCount = data[section].fault ? data[section].fault + sumCount : sumCount;
+          serious = data[section].serious ? !!data[section].serious : serious;
+          dangerous = data[section].dangerous ? !!data[section].dangerous : dangerous;
         }
```

Comment:
This looks like stuff that should eventually be in the store provider, it's fairly complex logic